config/config.go

```diff
@@ -0,0 +1,35 @@
+package config
+
+import (
+	"errors"
+	"io/ioutil"
+
+	validator "gopkg.in/validator.v2"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// errNoFilesToLoad is return when you attemp to call LoadFiles with no file paths
+var errNoFilesToLoad = errors.New("attempt to load configuration with no files")
+
+// LoadFiles loads a list of files, deep-merging values.
+// Validation is done after merging all values
+func LoadFiles(config interface{}, fnames ...string) error {
+	if len(fnames) == 0 {
+		return errNoFilesToLoad
+	}
+	for _, fname := range fnames {
+		data, err := ioutil.ReadFile(fname)
+		if err != nil {
+			return err
+		}
+
+		if err := yaml.Unmarshal(data, config); err != nil {
```

Comment:
I'm against merging unless it's absolutely necessary because it makes it difficult to reason about what the final values are for each config property without reading every config. It also means we now have one more config file we need to manage. AFAICT we've always been using a single config, so we can probably live without merging.

That said, I'm not against making the `LoadFiles` function unexported and have a `LoadFile` function that interally calls `LoadFiles` passing in a single file. That way if one day we do need the merging ability, we can easily export it.
  
Also, what are your thoughts on defining a simple mechanism to find default config files (e.g., always look inside the current working dir for `config.yaml`) and providing a function (e.g., `Load`) to load the default config file if it exists?